asset/generators.go

```diff
@@ -0,0 +1,319 @@
+package asset
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"math"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/lightninglabs/taproot-assets/internal/test"
+	"github.com/lightninglabs/taproot-assets/mssmt"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/tlv"
+	"pgregory.net/rapid"
+)
+
+const (
+	// MaxPrevWitnesses is the maximum number of PrevWitness objects
+	// that will be generated when generating []Witness.
+	MaxPrevWitnesses = 8
+
+	// MaxTxWitnessElementCount is the maximum number of elements in a
+	MaxTxWitnessElementCount = 64
+
+	// MaxTxWitnessElementSize is the maximum size of a single object in a
+	// generated witness stack.
+	MaxTxWitnessElementSize = 64
+)
+
+var (
+	// Simple generators.
+
+	VersionGen = rapid.SampledFrom([]Version{V0, V1})
```

Comment:
Nice! Was writing some tests with `rapid` in another PR and nearly reached to write custom gens for everything, but decided not to, but glad you did